### PR TITLE
Fix is_root check.

### PR DESCRIPTION
**Pull request by Neopallium**
Fix the `is_root` check in `utility.batch` extrinsic.

We are not using a `BaseCallFilter` right now, so I don't think this bug has any security issues.

### CODE REVIEW

File: pallets/utility/src/lib.rs [hunk 1]
```diff
@@ -180,7 +180,7 @@ decl_module! {
         /// event is deposited.
         #[weight = <T as Config>::WeightInfo::batch(&calls)]
         pub fn batch(origin, calls: Vec<<T as Config>::Call>) {
-            let is_root = Self::ensure_root_or_signed(origin.clone()).is_ok();
+            let is_root = Self::ensure_root_or_signed(origin.clone())?;
```
Comment (Centril) [inline pallets/utility/src/lib.rs:183]: Documentation above says "will return `Ok` in all circumstances

Comment (Neopallium) [reply pallets/utility/src/lib.rs:183]: It will only error if the transaction is not signed.

Comment (Neopallium) [reply pallets/utility/src/lib.rs:183]: I think the whole batch call should be rejected if `origin` is not signed.

Comment (Centril) [reply pallets/utility/src/lib.rs:183]: Can you update the docs to reflect this change?

Comment (Centril) [reply pallets/utility/src/lib.rs:183]: And some changelogs as well

Comment (Neopallium) [reply pallets/utility/src/lib.rs:183]: The docs are the same on all the `batch*` extrinsics.

Comment (Centril) [reply pallets/utility/src/lib.rs:183]: Then the docs are wrong on all:
> May be called from any origin.

But `ensure_none` is a valid origin not accepted.

**Review by Centril — APPROVED** [review]
Approved but let's get the additional docs changes we discussed on the meeting in.

**Review by adamdossa — APPROVED** [review]

**Pull request merged**